client/src/hooks/graphql/use-with-record-state.tsx

```diff
@@ -115,6 +123,38 @@ export function useWithRecordState(
     }
   }, [mentorError]);
 
+  useEffect(() => {
+    if (!mentor) return;
+    if (
+      recordPageState === RecordPageState.RELOADING_MENTOR &&
+      !isMentorLoading
+    ) {
+      setRecordPageState(RecordPageState.RECORDING_ANSWERS);
```

Comment:
this is the kind of thing that can potentially cause test flake. Calling `setRecordPageState` here could trigger a render where `recordPageState=RECORDING_ANSWERS` but whatever `nextAnswer` does has not occurred yet. Would that be valid or would it cause errors?

Probably react will batch this update before calling render (as long as they're synchronous), but it's a thing to watch out for